orttraining/orttraining/test/python/orttraining_test_ortmodule_fallback.py

```diff
@@ -554,10 +554,16 @@ def test_ortmodule_fallback_warn_message(is_training, persist_fallback):
     # Use data in different device for testing
     inputs = torch.randn(N, D_in, device=data_device)
 
-    for _ in range(3):
+    for i in range(3):
         with pytest.raises(RuntimeError), pytest.warns(UserWarning) as warning_record:
             ort_model(inputs)
-        assert "Fallback to PyTorch due to exception" in str(warning_record[0].message.args[0])
+
+        # Only log the warn message once or when retries is enabled
+        # Otherwise the warn message will be skipped
+        if i == 0 or not persist_fallback:
```

Comment:
I would just duplicate the call so that there is not conditionals in tests (error prone)

Reply:
Hi Justin, thank you so much for your review. 

If the `retries` (persist_fallback=True) is disabled, the warning message will be logged only once and skipped for the rest. However, if the `retries` (persist_fallback=False) is enabled, the warning message will be logged in each loop.

Can you please share more details about `duplicate the call`?